internal/gatewayapi/contexts.go

```diff
@@ -446,6 +446,88 @@ func (u *UDPRouteContext) GetRouteParentContext(forParentRef v1beta1.ParentRefer
 	return ctx
 }
 
+// TCPRouteContext wraps a TCPRoute and provides helper methods for
+// accessing the route's parents.
+type TCPRouteContext struct {
+	*v1alpha2.TCPRoute
+
+	parentRefs map[v1beta1.ParentReference]*RouteParentContext
+}
+
+func (u *TCPRouteContext) GetRouteType() string {
+	return KindTCPRoute
+}
+
+// GetHostnames return empty string array because TCPRoute has no hostnames
```

Comment:
does this method have to be added ?

Reply:
Same as:

```go
// GetHostnames return empty string array because UDPRoute has no hostnames
func (u *UDPRouteContext) GetHostnames() []string {
	return []string{""}
}
```

Reply:
If we need to remove this, we should clean UDPRoute method too.

Reply:
And this is a RouteContext interface, if we do not implement it, there will be reported errors:

```go
// RouteContext represents a generic Route object (HTTPRoute, TLSRoute, etc.)
// that can reference Gateway objects.
type RouteContext interface {
	client.Object

	// GetRouteType returns the Kind of the Route object, HTTPRoute,
	// TLSRoute, TCPRoute, UDPRoute etc.
	GetRouteType() string

	// TODO: [v1alpha2-v1beta1] This should not be required once all Route
	// objects being implemented are of type v1beta1.
	// GetHostnames returns the hosts targeted by the Route object.
	GetHostnames() []string

	// TODO: [v1alpha2-v1beta1] This should not be required once all Route
	// objects being implemented are of type v1beta1.
	// GetParentReferences returns the ParentReference of the Route object.
	GetParentReferences() []v1beta1.ParentReference

	// GetRouteParentContext returns RouteParentContext by using the Route
	// objects' ParentReference.
	GetRouteParentContext(forParentRef v1beta1.ParentReference) *RouteParentContext
}
```

Reply:
need to open an ticket track this

Reply:
We should consider if we need to remove this in RouteContext. It seems like a common method for L7 Routes but not for L4 Routes.

```go
	// TODO: [v1alpha2-v1beta1] This should not be required once all Route
	// objects being implemented are of type v1beta1.
	// GetHostnames returns the hosts targeted by the Route object.
	GetHostnames() []string
```

Reply:
I am removing this.